src/value_generators/test/expression_test.cpp

```diff
@@ -238,6 +238,13 @@ TEST_CASE("Expression parsing with ConstantExpression::parse") {
     genny::DefaultRandom rng{};
     rng.seed(269849313357703264LL);
 
+    SECTION("recursive application caught at setup time") {
```

Comment:
Is this so much of being recursive as just being the wrong type?

Reply:
This is proving that the type-checking is done recursively at ctor time not at runtime. This didn't used to fail until `evaluate()` was called.